lib/getUserConstructor.js

```diff
@@ -0,0 +1,268 @@
+const ms = require('ms');
+
+/**
+ * @param {AuthenticationPlugin} plugin
+ * @returns {User}
+ */
+module.exports = plugin => {
+  // private properties need to be hidden from serialization
+  const _kuzzleUser = Symbol.for('_kuzzleUser');
```

Comment:
You can use the [Reflect.defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/defineProperty) method to hide property from serialization:
```
Reflect.defineProperty(this, '_kuzzleUser', {
  writable: true
  // "enumerable" is "false" by default
});
```

Reply:
:+1: , much more natural indeed.